File: backend/src/main/java/com/cesarschool/forjaapi/models/Arma.java
package com.cesarschool.forjaapi.models;

public class Arma extends Item {
    private float dano;
    private String tipo;

    public Arma() {
        super();
    }

    public Arma(int id, String nome, float valor, float peso, String raridade, float dano, String tipo) {
        super(id, nome, valor, peso, raridade);
        this.dano = dano;
        this.tipo = tipo;
    }

    public float getDano() {
        return dano;
    }

    public String getTipo() {
        return tipo;
    }

    public void setDano(float dano) {
        this.dano = dano;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }
}

File: frontend/src/models/Arma.ts
export const Arma = {
    nome: "arma",
    atributos: [
      { nome: "nome", tipo: "string" },
      { nome: "valor", tipo: "number" },
      { nome: "peso", tipo: "number" },
      { nome: "dataFabricacao", tipo: "string" },
      { nome: "raridade", tipo: "string" },
      { nome: "ferreiro", tipo: "foreign", referencia: "ferreiro" },
      { nome: "dano", tipo: "number" },
      { nome: "tipo", tipo: "string" }
    ]
};